ui/js/controllers/logviewer.js

```diff
@@ -41,6 +41,52 @@ logViewerApp.controller('LogviewerCtrl', [
             $scope.showSuccessful = !$scope.hasFailedSteps();
         });
 
+        $scope.$watch('[selectedBegin, selectedEnd]', function(newVal, oldVal) {
+            var newHash = (newVal[0] === newVal[1]) ? newVal[0] : newVal[0] + "-L" + newVal[1];
+            if (!isNaN(newVal[0])) {
+                $location.hash("L" + newHash);
+            } else {
+                if ($scope.artifact) {
```

Comment:
I just realized, could this nested if statement be converted to an `} else if ($scope.artifact) {` on line 48.

Let me know if that makes sense to you, and if in testing it's working correctly.